### PR TITLE
ran git reset --hard origin/master -> 9f139c, made the shorthand chan…

### PR DESCRIPTION
**Pull request by kevintriestocode**
i think i figured it out.
Just wanted to pull most recent version from github and change ONLY 3 lines of code

from **master**, I ran
`git reset --hard origin/master`
    changed the 3 lines of code
`git commit -a -m "message"`
`git checkout -b more-helloworld-edits`

Here, I ran into an error:
`git push --set-upstream origin more-helloworld-edits`

> To github.com:Ryxias/chuuni.git
> ! [rejected]        more-helloworld-edits -> more-helloworld-edits (non-fast-forward)
> error: failed to push some refs to 'git@github.com:Ryxias/chuuni.git'
> hint: Updates were rejected because the tip of your current branch is behind
> hint: its remote counterpart. Integrate the remote changes (e.g.
> hint: 'git pull ...') before pushing again.
> hint: See the 'Note about fast-forwards' in 'git push --help' for details.

So I instead ran:
`git push origin more-helloworld-edits --force`

Is there a difference between the above ^ and with an additional `/`?
ex. `git push origin/more-helloworld-edits --force`

### CODE REVIEW

**Comment by Ryxias** [comment]
There is a subtle difference between the branches `origin/more-helloworld-edits` and `more-helloworld-edits` and `origin more-helloworld-edits`.

TL;DR...

#### `more-helloworld-edits`
This is your branch, on your local git repo. You create this branch when you do 

```
$ git checkout -b more-helloworld-edits
```

#### `origin more-helloworld-edits`
This means the `more-helloworld-edits` ON **origin**, NOT on your local git repo. You first create this branch when you push TO origin:

```
$ git push origin more-helloworld-edits
```

The above command is actually short-hand for:

```
$ git push origin more-helloworld-edits:more-helloworld-edits
```

Which basically means "push my local `more-helloworld-edits` onto origin".


#### `origin/more-helloworld-edits`
This is a *reference* to a origin's `more-helloworld-edits` branch, **on your local git repo**.

The reason why this is special is because, this is how your git repo tracks origin's branches. This is **where your repo thinks origin's `more-helloworld-edits` branch is at**. Every time you do:

```
$ git fetch
```

What you're actually doing is updating your `origin/more-helloworld-edits` reference with the true value of origin's `more-helloworld-edits` branch (if it has changed). This is most common with `origin/master`.


There's actually nothing special about the slash `/`. The name `origin/xxxxyyyzzz` is a perfectly valid branch name. For example, you can `git checkout -b origin/balh/blahbl/blasdlfajlf` and it will totally let you do that (just... DONT do that).
